pandas/core/generic.py

```diff
@@ -469,9 +469,20 @@ def transpose(self, *args, **kwargs):
         if kwargs.pop('copy', None) or (len(args) and args[-1]):
             new_values = new_values.copy()
 
-        if kwargs:
-            raise TypeError('transpose() got an unexpected keyword '
-                            'argument "{0}"'.format(list(kwargs.keys())[0]))
+        try:
+            nv.validate_transpose(tuple(), kwargs, method='kwargs')
```

Comment:
why is this all here? just pass ``self`` to all of these validators and you can do this in side the compat. This is just too much inside this method.